agent/agent_test.go

```diff
@@ -723,18 +728,7 @@ func (r *AgentSuite) becomeActiveMember(membership *mockClusterMembership, agent
 	return r.becomeMember(membership, agent, MemberAlive)
 }
 
-// func (r *AgentSuite) becomeInactiveMember(membership *mockClusterMembership, agent *agent) error {
-// 	return r.becomeMember(membership, agent, MemberLeft)
-// }
-
 func (r *AgentSuite) becomeMember(membership *mockClusterMembership, agent *agent, status MemberStatus) error {
-	if _, ok := membership.members[agent.Name]; ok {
```

Comment:
Not sure I followed the logic here and why this check was in place? If the intention is to update the member's status (which I think was the primary goal), then it's irrelevant. Otherwise, I think it would be better to split creating an agent (`newAgent`) and adding a member (now `newAgent` creates an active member explicitly). @bernardjkim Let me know if I misunderstood this piece.